lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -0,0 +1,158 @@
+use std::sync::{Arc, Mutex, OnceLock};
+
+use pyo3::prelude::*;
+use saluki_error::{generic_error, GenericError};
+use tokio::sync::mpsc::Sender;
+use tracing::{debug, error, trace};
+
+use crate::sources::checks::check_metric::{CheckMetric, MetricType};
+
+// Global state to store the sender
+static METRIC_SENDER: OnceLock<Arc<Mutex<Sender<CheckMetric>>>> = OnceLock::new();
```

Comment:
You still don't need a `Mutex` for this.

All the sender methods on `Sender` are already safe to use from multiple callers (`&self`) and the initialization of `METRIC_SENDER` itself is already synchronized by `OnceLock`.